internal/cluster/validations/validations.go

```diff
@@ -312,73 +315,298 @@ func ValidateVipDHCPAllocationWithIPv6(vipDhcpAllocation bool, machineNetworkCID
+			multiErr = multierror.Append(multiErr, errors.Errorf("Could not parse VIP ip %s", string(ingressVips[i].IP)))
 		}
-		targetConfiguration.ClusterNetworks = c.ClusterNetworks
-		targetConfiguration.ServiceNetworks = c.ServiceNetworks
-		targetConfiguration.MachineNetworks = c.MachineNetworks
+	}
+	if multiErr != nil && !strings.Contains(multiErr.Error(), "0 errors occurred") {
+		return multiErr
 	}
 
-	allAddresses = append(allAddresses, swag.String(ingressVip), swag.String(apiVip))
-	allAddresses = append(allAddresses, common.GetNetworksCidrs(obj)...)
+	return nil
+}
 
-	err := ValidateIPAddressFamily(ipV6Supported, allAddresses...)
+func validateIPAddressesInput(targetConfiguration common.Cluster) error {
+	if len(targetConfiguration.APIVips) > 2 {
+		return errors.Errorf("api_vips supports 2 vips. got: %d", len(targetConfiguration.APIVips))
+	}
+	if len(targetConfiguration.IngressVips) > 2 {
+		return errors.Errorf("ingress_vips supports 2 vips. got: %d", len(targetConfiguration.IngressVips))
+	}
+
+	if err := VerifyParsableVIPs(targetConfiguration.APIVips, targetConfiguration.IngressVips); err != nil {
+		return err
+	}
+
+	if len(targetConfiguration.APIVips) != len(targetConfiguration.IngressVips) {
+		return errors.Errorf("configuration must include the same number of api_vips (got %d) and ingress_vips (got %d)",
+			len(targetConfiguration.APIVips), len(targetConfiguration.IngressVips))
+	}
+	return nil
+}
+
+func validateINetworksIPAddressFamily(ipV6Supported bool, targetConfiguration common.Cluster) error {
+	var (
+		networks []*string
+		err      error
+		multiErr error
+	)
+
+	machineNetworks := network.DerefMachineNetworks(funk.Get(targetConfiguration, "MachineNetworks"))
+	serviceNetworks := network.DerefServiceNetworks(funk.Get(targetConfiguration, "ServiceNetworks"))
+	clusterNetworks := network.DerefClusterNetworks(funk.Get(targetConfiguration, "ClusterNetworks"))
+
+	for i := range machineNetworks {
+		networks = append(networks, swag.String(string(machineNetworks[i].Cidr)))
+	}
+	err = ValidateIPAddressFamily(ipV6Supported, networks...)
+	multiErr = multierror.Append(multiErr, err)
+	for i := range serviceNetworks {
+		networks = append(networks, swag.String(string(serviceNetworks[i].Cidr)))
+	}
+	err = ValidateIPAddressFamily(ipV6Supported, networks...)
+	multiErr = multierror.Append(multiErr, err)
+	for i := range clusterNetworks {
+		networks = append(networks, swag.String(string(clusterNetworks[i].Cidr)))
+	}
+	err = ValidateIPAddressFamily(ipV6Supported, networks...)
+	multiErr = multierror.Append(multiErr, err)
+
+	if multiErr != nil && !strings.Contains(multiErr.Error(), "0 errors occurred") {
+		return multiErr
+	}
+	return nil
+}
+
+func validateVips(targetConfiguration common.Cluster) ([]*string, error) {
+	var allAddresses []*string
+	var err error
+
+	if len(targetConfiguration.APIVips) == 1 {
+		allAddresses = append(allAddresses, swag.String(string(targetConfiguration.APIVips[0].IP)))
```

Comment:
It may be covered already in some other part of code, but for 2 VIPs you do a validation that 1st one is V4 and 2nd one is V6. Here, for a single VIP scenario, you just append without a validation. Any reason? To make it look consistent I would consider
```
if len(targetConfiguration.APIVips) == 1 {
  	if !network.IsIPv4Addr(string(targetConfiguration.APIVips[0].IP)) {
		err = errors.Errorf("the first element of api_vips must be an IPv4 address. got: %s", targetConfiguration.APIVips[0].IP)
		return nil, err
	}
	allAddresses = append(allAddresses, swag.String(string(targetConfiguration.APIVips[0].IP)))
}
```

Reply:
added